### PR TITLE
Feature/SDL passenger mode

### PR DESCRIPTION
**Pull request by HSavynetska**
ATF Test Scripts to check #[2134](https://github.com/smartdevicelink/sdl_core/issues/2134)

This PR is **ready** for review.

### Summary
Scripts for "SDL Passenger Mode" feature

### ATF version
[develop](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by GetmanetsIrina** [comment]
@jacobkeeler, the update according to review comments is performed, please review in  https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2160/commits/4baee3c85d850c5d8f9c120a7a798d8cacbcf744

File: user_modules/sequences/actions.lua [hunk 1]
```diff
@@ -137,15 +137,19 @@ end
...
+  event.matches = function(e1, e2) return e1 == e2 end
+  m.getHMIConnection():SendNotification("SDL.OnAllowSDLFunctionality", {
     allowed = true,
     source = "GUI",
     device = {
       id = utils.getDeviceMAC(),
       name = utils.getDeviceName()
     }
   })
+  RUN_AFTER(function() m.getHMIConnection():RaiseEvent(event, "Allow SDL event") end, 500)
```
Comment (jacobkeeler) [inline user_modules/sequences/actions.lua]: Is 500ms necessary? I could see that causing a lot of overhead, given that this is part of every test.

**Comment by jacobkeeler** [comment]
Another note, it seems that the smoke tests fail on https://github.com/smartdevicelink/sdl_core/pull/2783 with:
```
[11:25:25,223] RAI                                                                                   [FAIL] (10871 ms)
 OnDriverDistraction notification: Timeout: OnDriverDistraction notification: Timeout expired
```
 I don't know whether this is a script or Core issue.

File: user_modules/sequences/actions.lua (hunk 1 above)
Comment (dboltovskyi) [reply user_modules/sequences/actions.lua]: @jacobkeeler Unfortunately this timeout is required for SDL in order it has enough time to recalculate device permissions. (applicable for SDL in `EXTERNAL_PROPRIETARY` mode). We could try to reduce this number to a lower value (e.g. for 100-200ms)

Comment (dboltovskyi) [reply user_modules/sequences/actions.lua]: @jacobkeeler We've just tried to run the tests with 250ms timeout and aprox. 10% of runs failed. So I propose to leave 500ms to have more stable runs.
Taking into account that average simple test script is executed during 10 sec. we will have only 5% increase with 500ms delay.

**Comment by dboltovskyi** [comment]
@jacobkeeler We're unable to reproduce this issue on our environment.
Is it reproduced with every test scripts from PR or just with a single one?
Is it reproduced randomly?
Could you provide SDL and ATF logs please?
One of the reason could be not correct `sdl_preloaded_pt.json` file which is updated during script execution.
Could you restore it from initial SDL build and re-run the tests?

**Comment by jacobkeeler** [comment]
@dboltovskyi Don't know what the cause of this was, but a clean rebuild did appear to fix the issue. Approving.

**Comment by IGapchuk** [comment]
@jacobkeeler @JackLivio Please, be informed that we've added new commits that cover implementation for next revisions of the proposal:  smartdevicelink/sdl_evolution#766 and smartdevicelink/sdl_evolution#775. Please take a look at next commits: d355522dac1f24b7fac4a1afc3da6cb934e928fc, a2052be3d2402480bfbe476214330ed4a3974efb, 2c2c74223bc1342a57ff813707ae6e6e05bc0672, aa087830148310bc514334162131ab58a4f58d00, a5c5f0ddcbe7421d49650dc4c291984fd232116e, 1229a975ef9d11e6d837fb6193f600a21eac0090

**Comment by mrapitis** [comment]
@theresalech Ford has reviewed and approved this PR.

**Pull request merged**